### DISCUSSION TITLE
How do I need to "tag" a document region to enable lemminx dependency auto completion?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse/lemminx-maven#532] [Q&A]
In m2e we have an extension for the PDE target editor.

The serialized form is (mostly) what maven uses for example it looks like this:
https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/520a195d09f2adc6f2caca84a7087ad96fa8c43f/eclipse.platform.releng.prereqs.sdk/eclipse-sdk-prereqs.target#L99-L111

What I like to archive now is:

1. Mark the reqion as a custom content type, this seem to be related to the client side
2. Then I need to connect that region to be handled by the  lemminx language server so it suggest new dependencies/versions/auto completion like in pom.xml editor

It would be great if some of the lemminx / WWD / ... devs could help here ...

**Comment by mickaelistria** [comment]
> Mark the reqion as a custom content type, this seem to be related to the client side

The LS isn't aware of client-side content types for text regions.

I believe the best approach is to relax lemminx-maven so it can support the case of dependency/dependencies nodes even if the files aren't pom files. If there are no strong risk of overlap (ie the tags used in .target don't match the ones used by poms, except for the ones where they should behave the same; then you might try to relax the case of https://github.com/eclipse/lemminx-maven/blob/master/lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/MavenLemminxExtension.java#L584 to support .target in some extent (although many features assuming that the file can be parsed as a pom may fail more or less gracefully)
